src/user-profile.js

```diff
@@ -140,6 +153,30 @@ const Profile = {
 
       fs.rmdirSync(oldProfileDir);
     }
+
+    // Move add-ons.
+    if (process.env.NODE_ENV !== 'test') {
+      const oldAddonsDir = path.join(this.gatewayDir, 'build', 'addons');
+      if (fs.existsSync(oldAddonsDir) &&
+          fs.lstatSync(oldAddonsDir).isDirectory()) {
+        const fnames = fs.readdirSync(oldAddonsDir);
+        for (const fname of fnames) {
+          const oldFname = path.join(oldAddonsDir, fname);
+          const newFname = path.join(this.addonsDir, fname);
+          const lstat = fs.lstatSync(oldFname);
+
+          if (fname !== 'plugin' && lstat.isDirectory()) {
+            // Move existing add-ons.
+            fs.renameSync(oldFname, newFname);
+          } else if (fname.endsWith('.js') && lstat.isFile()) {
+            // Symlink dependencies.
+            if (!fs.lstatSync(newFname).isSymbolicLink()) {
+              fs.symlinkSync(oldFname, newFname);
+            }
+          }
+        }
+      }
+    }
```

Comment:
Rather than create symlinks, would it make sense to move the entire tree using something like there move command from the fs-extra package.

Reply:
I wanted to keep the js files in the build tree such that when they're updated in the source tree, we don't have to md5 or anything to figure out if we need to copy again. It also didn't feel right putting source that is imported by both the gateway and add-ons in the user profile.

Ideally, this will all go away in the future after we make a node package for the add-ons to import.